### DISCUSSION TITLE
mass difference cause objects to clip

### DISCUSSION THREAD

**Discussion by albinaask** [jrouwe/JoltPhysics#1394] [Q&A]
I've used Godot engine with the godot-Jolt addon to use as a test environment, but I suspect the bug is on Jolt's side of things.

The floor is a static body.

On that two cubes are falling, one with the mass of 1 kg and the top one with a mass of 100kg. The expected behaviour would be the exact same as in the case of ordinary falling cubes, aka that they stack neatly, which is the case when they both have a mass of 1kg.  

https://github.com/user-attachments/assets/8dffc930-157b-427f-a0be-ff7707778140

**Comment by jrouwe** [comment]
This is an artifact of the solver used (and is a problem with many physics engines), it cannot solve very heavy objects falling on very light objects. The only way you improve this is by increasing the number of solver iterations (called 'Velocity Steps'/'Position Steps' in godot's 'Jolt Physics 3D' settings).
